aviatrix/resource_aviatrix_firewall_instance.go

```diff
@@ -334,6 +341,12 @@ func resourceAviatrixFirewallInstanceCreate(d *schema.ResourceData, meta interfa
 		}
 	}
 
+	if tags, ok := d.GetOk("tags"); ok && (cloudType == goaviatrix.AWS || cloudType == goaviatrix.AWSGOV || cloudType == goaviatrix.AZURE) {
+		firewallInstance.Tags = tags.(map[string]interface{})
```

Comment:
Could we just use map[string]string instead? I think the map[string]interface{} is a bit confusing.

I think the best path forward is a function like `func getTags(d d *schema.ResourceData) map[string]string` that will pull out the "tags" attribute and convert to map[string]string. You could also put the colon escaping logic in this function so that we can reuse it later when we change all `tag_list` to `tags`